packages/python/src/armonik/common/filter.py

```diff
@@ -0,0 +1,443 @@
+        """
+            Transforms the filter into a disjunction
+        Returns:
+            Equivalent disjunction of this filter
+        """
+        raise NotImplementedError(f"{str(self.__class__.__name__)}.to_disjunction() is not implemented")
+
+
+class FilterDisjunction(Filter):
+    """
+        Represents a disjunction of filters (logical or)
+    """
+    def __init__(self, filters: List["FilterConjunction"]):
+        super().__init__()
+        self.filters = filters
+        _ = self.conjunction_type()
+        _ = self.message_type()
+
+    def __or__(self, other: Union["SimpleFilter", "FilterConjunction", "FilterDisjunction"]) -> "FilterDisjunction":
+        """
+            Returns a new disjunction from the operands
+        Args:
+            other:
+                Other filter or filters
+        Returns:
+            New disjunction
+        """
+        if isinstance(other, SimpleFilter):
+            # Create a conjunction from the filter then add to the list
+            if other.conjunction_type != self.conjunction_type():
+                raise Exception(f"Invalid type {type(other).__name__} ({str(other.field)}) for 'or' operand of {self.__class__.__name__} ({str(self)}) : Conjunction types are different")
+            return self.__class__(self.filters + [self.conjunction_type()([other])])
+        elif isinstance(other, self.conjunction_type()):
+            # Add the conjunction to the list
+            return self.__class__(self.filters + [other])
+        elif isinstance(other, self.__class__):
+            # Fuse the disjunctions
+            return self.__class__(self.filters + other.filters)
+        raise Exception(f"Invalid type {type(other).__name__} for 'or' operand of {self.__class__.__name__}")
+
+    def __add__(self, other: Union["SimpleFilter", "FilterConjunction", "FilterDisjunction"]) -> "FilterDisjunction":
+        """
+            Equivalent to __or__
+        """
+        return self | other
+
+    def __repr__(self):
+        return "( " + ' ) or ( '.join([str(f) for f in self.filters]) + ' )'
+
+    def conjunction_type(self) -> Type["FilterConjunction"]:
+        """
+            Conjunction type associated with this disjunction type.
+            Must be overriden by child class
+        Returns:
+            Type of FilterConjunction
+        """
+        raise NotImplementedError(f"{str(self.__class__.__name__)}.conjunction_type() is not implemented")
+
+    def to_message(self) -> Message:
+        raw = self.message_type()()
+        # Can't use raw.or : https://protobuf.dev/reference/python/python-generated/#keyword-conflicts
+        getattr(raw, "or").extend([f.to_message() for f in self.filters])
+        return raw
+
+    def to_conjunction(self) -> "FilterConjunction":
+        raise Exception("Cannot transform a disjunction into a conjunction")
+
+    def to_disjunction(self) -> "FilterDisjunction":
+        return self
+
+
+class FilterConjunction(Filter):
```

Comment:
I think these should not inherit `Filter`

Reply:
They are at a higher level in the AST

Reply:
Filter is more of an interface in this case. So it wouldn't be that strange imo.